activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -138,9 +138,14 @@ def create_all
       end
 
       def setup_initial_database_yaml
-        return {} unless defined?(Rails)
+        return {} if !defined?(Rails) || Rails.application.config.paths["config/database"].existent.empty?
 
-        Rails.application.config.load_database_yaml
+        begin
+          Rails.application.config.singleton_class.define_method(:method_missing) { |selector, *args, &blk| self }
```

Comment:
I don't see why the rails framework would essentially monkey patch an object it owns. I don't think this is a good solution.

Reply:
Rails already does that, it's just more elegant to monkeypatch than to replace that object: https://github.com/rails/rails/blob/main/railties/lib/rails/application/configuration.rb#L388-L403

Reply:
It isn't. It is more intrusive, prone to problems and confusing in this case. Monkey patches can be elegant, but it doesn't mean they are easier to understand than swapping an object to something people can look in the implementation

Reply:
Fair point. I think we could still improve by using `ActiveSupport::ConfigurationFile.parse` and not doing all that in the `load_database_yaml` manually. Would you accept a PR for that?

Reply:
Yes. I think that makes sense.

Reply:
Like this? https://github.com/rails/rails/pull/46161/files